saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/process/UploadEnrichmentTagOverwriterTest.java

```diff
@@ -0,0 +1,146 @@
+    
+    @Test
+    public void shouldEnrichmentTagsBeOverwritten_false_uploadTagMaxDate_isNull() {
+        
+        // given
+        
+        when(uploadEnrichmentTagRepository.findMaxCreationDate()).thenReturn(null);
+        
+        // execute
+        
+        boolean shouldEnrichmentTagsBeOverwriten = uploadEnrichmentTagOverwriter.shouldEnrichmentTagsBeOverwritten();
+        
+        // assert
+        
+        assertFalse(shouldEnrichmentTagsBeOverwriten);
+        verify(enrichmentTagRepository, never()).findMaxCreationDate();
+        
+    }
+    
+    
+    @Test
+    public void shouldEnrichmentTagsBeOverwritten_true_uploadTagMaxDate_isNotNull_tagMaxDate_isNull() {
+        
+        // given
+        
+        when(uploadEnrichmentTagRepository.findMaxCreationDate()).thenReturn(new DateTime(2012, 12, 12, 12, 13));
+        when(enrichmentTagRepository.findMaxCreationDate()).thenReturn(null);
+        
+        // execute
+        
+        boolean shouldEnrichmentTagsBeOverwriten = uploadEnrichmentTagOverwriter.shouldEnrichmentTagsBeOverwritten();
+        
+        // assert
+        
+        assertTrue(shouldEnrichmentTagsBeOverwriten);
+        
+    }
+    
+    
+    @Test
+    public void shouldEnrichmentTagsBeOverwritten_true_uploadTagMaxDate_isNotNull_tagMaxDate_isBefore() {
+        
+        // given
+        
+        when(uploadEnrichmentTagRepository.findMaxCreationDate()).thenReturn(new DateTime(2012, 12, 12, 12, 13));
+        when(enrichmentTagRepository.findMaxCreationDate()).thenReturn(new DateTime(2012, 12, 12, 12, 11, 23));
+        
+        // execute
+        
+        boolean shouldEnrichmentTagsBeOverwriten = uploadEnrichmentTagOverwriter.shouldEnrichmentTagsBeOverwritten();
+        
+        // assert
+        
+        assertTrue(shouldEnrichmentTagsBeOverwriten);
+        
+    }
+    
+    
+    @Test
+    public void shouldEnrichmentTagsBeOverwritten_true_uploadTagMaxDate_isNotNull_tagMaxDate_isAfter() {
+        
+        // given
+        
+        when(uploadEnrichmentTagRepository.findMaxCreationDate()).thenReturn(new DateTime(2012, 12, 12, 12, 13));
+        when(enrichmentTagRepository.findMaxCreationDate()).thenReturn(new DateTime(2012, 12, 13, 12, 11, 23));
+        
+        // execute
+        
+        boolean shouldEnrichmentTagsBeOverwriten = uploadEnrichmentTagOverwriter.shouldEnrichmentTagsBeOverwritten();
+        
+        // assert
+        
+        assertFalse(shouldEnrichmentTagsBeOverwriten);
+        
+        
+    }
+    
+    
+    @Test
+    public void overwriteEnrichmentTags() {
+
+
+        // execute
+        
+        uploadEnrichmentTagOverwriter.overwriteEnrichmentTags();
+        
+        // assert
+        
+        verify(enrichmentTagRepository).deleteAll();
+        verify(uploadEnrichmentTagCopier).copyUploadedEnrichmentTags();
+        verify(uploadEnrichmentTagRepository, never()).deleteAll();
```

Comment:
Why you check here for `verify(uploadEnrichmentTagRepository, never()).deleteAll();`? Is it a precaution, that in the future no one should try to remove uploadedEnrichmentTags after they are copied? Or is it just to show that uploadedET are not removed after copying process?